galaxyproject/galaxy
# Overhaul Galaxy Markdown Help

There was a single help document for both pages and invocation reports even though the details of how to use the directives were quite different. The resulted in a lot of incorrect and un-useful help text depending on how you got to the component.

This component now dispatches on the Markdown document type and generates correct examples and correct help text depending on this (prop called 'mode'). The help text for the individual commands is now pulled out of the YAML document introduced in #16979 and that is also used for the mouseover help in the side panel for the directives. The process of synchronizing this resulted in more complete documentation for the commands/directives in both places.

I'm including before and after screenshots for the workflow invocation report help because I think it was the more broken of the views - but both views have been improved.

Invocation Help On Main (Pretty Broken):

<img width="731" alt="Screenshot 2023-12-22 at 2 39 01 PM" src="https://github.com/galaxyproject/galaxy/assets/216771/0ac2955f-d015-4e15-a29e-e018228d78e8">

<img width="818" alt="Screenshot 2023-12-22 at 2 39 49 PM" src="https://github.com/galaxyproject/galaxy/assets/216771/e6171690-93bb-450e-adc1-154277f08a84">

Invocation Help After PR (Pretty Okayish):

<img width="754" alt="Screenshot 2023-12-22 at 2 39 39 PM" src="https://github.com/galaxyproject/galaxy/assets/216771/3697bfcc-14ff-47df-af78-24440874338a">

<img width="807" alt="Screenshot 2023-12-22 at 2 39 57 PM" src="https://github.com/galaxyproject/galaxy/assets/216771/2786d96b-bb90-4a02-819c-aa34aca11128">

No longer building on unmerged PRs, but builds heavily on the merged xref #16979.

## How to test the changes?
(Select all options that apply)
  1. Look at the help sections for the respective components and see they aren't broken. 

## License
